### ISSUE TITLE
Missing changelog / suggest creating a specific upgrade guide.

### ISSUE THREAD

**Issue by JonRowe** [phoenixframework/phoenix_view#16]
:wave: Thanks for all your work, from investigating the changes in version 2 I noticed that there is no changelog entry, it'd be great if there could be one put together with all the changes, it'd also be nice if there was a seperate pulled out upgrade guide for the docs, so its easy to find the info on how to upgrade from 1.x to 2.x.

**Comment by josevalim** [comment]
We have pushed the CHANGELOG to git yesterday. The upgrade notes are in the docs for `Phoenix.View` but it should be a clean upgrade except for the fact Phoenix.Template is now a separate package (which shouldn't really break either).

**Comment by JonRowe** [comment]
@josevalim Apologies but where? https://github.com/phoenixframework/phoenix_view/blob/master/CHANGELOG.md Still is missing 2.x

**Comment by josevalim** [comment]
Gah, I could swear I pressed the merge button yesterday for #14 but I didn't. I have done that now. :)

In any case, I want to further clarify that the library should be backwards compatible. You still use it the same. Phoenix v1.7 won't use this library anymore (and that's up to Phoenix) but existing users should be completely fine.

**Comment by JonRowe** [comment]
Ah thank you! I mention the guide because the instructions didn't jump out at me from the docs, and it seemed like this would be a quite comprehensive change to Phoenix's "way of working" I guess prehaps I should wait for Phoenix 1.7 to have upgrade instructions.

**Comment by josevalim** [comment]
Right, I don't think there is anything to change until Phoenix v1.7 is out. Even then, I would wait until the final version is out, because we may change things a bit while in release candidate.

So my suggestion would be to generate a new app when the RC is out and play with things to get a feeling (and give us feedback!). Note the changes are only to the generated apps too, not in the APIs, so there is nothing forcing you to migrate either!

**Comment by JonRowe** [comment]
This suggestion was triggered by conversations in the elixir slack where some people have upgraded "because its there", get deprecation notices and are naturally driven to upgrade/change the existing code, so it was aimed at "easing" that for people

**Comment by josevalim** [comment]
@JonRowe Oh I see, thanks for sharing more context. I believe the deprecation notices it emits today are accurate though:

* `render_layout` => use components with slots
* `render_existing` => use function_exported?

Those should be doable and the tools to address them already exist. So I assume the question is how to do those to?

**Comment by JonRowe** [comment]
Maybe just examples and more linking to #module-migrating-to-phoenix-component ?

**Comment by josevalim** [comment]
I will add an example to the docs of those functions. :)

**Comment by josevalim** [comment]
I shipped v2.0.1 without the deprecation, so there is no push for urgency, and with better docs.

**Comment by JonRowe** [comment]
Thanks once again for your incredible care and attention to detail 👏
